addons/payment/controllers/portal.py

```diff
@@ -127,13 +127,23 @@ def payment_pay(
         # Generate a new access token in case the partner id or the currency id was updated
         access_token = payment_utils.generate_access_token(partner_sudo.id, amount, currency.id)
 
+        # First verify the existence of the invoice and second set the amount to zero in case
+        # of a cancelled invoice.
```

Comment:
Prefer to explain the why than to describe your code. Here is an example of what your comment could be:
```suggestion
        # If the invoice is cancelled, the payment link still allows to go through the payment. 
        # This code ensures that no payment can be made when the state of the invoice is cancelled.
```

Reply:
done :)